frontend/src/controllers/EditCampaignController.es6

```diff
@@ -0,0 +1,21 @@
+export default class {
+
+    /* @ngInject */
+    constructor(campaignService, $stateParams, $scope) {
+        this.service = campaignService;
+        this.$scope = $scope;
+        this.fetchCampaign($stateParams.code);
+    }
+
+    fetchCampaign(code) {
+        let campaign = this.service.all();
+        campaign.then(r => r.filter(v => v.code == code)[0]).then(r => {
+            this.$scope.campaign = r ? r : {name: "", code: ""}
+        });
+    }
+
+    save(campaign) {
+        this.service.save(campaign);
+        console.log(campaign, "is saved now");
```

Comment:
Reminder to remove this logging, or to raise a Trello ticket to cover creating a proper logging strategy.